### PR TITLE
builtin/aws: add ability to pass additional security groups

### PR DESCRIPTION
**Pull request by psihachina**
This commit adds the ability to pass additional security groups, for example
```
app "vole" {
 
  ...
  ...
  ...

  deploy {
    use "aws-ecs" {
      region = "us-east-2"
      memory = "512"
      cluster = "dev-nutcorp"
      security_groups = ["sg-0a98e407efe1aa683", "sg-078fc39329784c141"]
    }
  }
}
```
image

### Test
[asciicast](https://asciinema.org/a/xloZv22H6EkC44aFdefx3UzJ4)

### CODE REVIEW

**Comment by krantzinator** [comment]
What do you think about providing a replacement set of security groups, rather than adding on to the default that Waypoint sets? So instead of `append` you'd just make a new slice.
Would this solve the root problem?

(Please remove the `waypoint-examples` submodule again 🙂 )

File: .changelog/1740.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:feature
+builtin/aws: add ability to pass additional security groups for ecs
```
Comment (briancain) [inline .changelog/1740.txt:2]:
```suggestion
```release-note:improvement
plugin/aws: add ability to pass additional security groups for ecs
```

**Review by briancain — CHANGES_REQUESTED** [review]
+1 to what @krantzinator mentioned.

We should fix up the changelog here too. We keep new functionalities as `feature`, where as new additions or improvements to existing plugins should be listed as `improvement`. We also opt for naming the changelog entry with `plugin` over `builtin` since its more obvious what that means to a user reading the CHANGELOG not familiar with the waypoint repo structure. Thanks!

File: .changelog/1740.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+plugin/aws: add ability to pass additional security groups for ecs
```
Comment (evanphx) [inline .changelog/1740.txt:2]:
```suggestion
plugin/aws: add ability to define existing security groups for ecs
```

File: builtin/aws/ecs/platform.go [hunk 3]
```diff
@@ -1567,6 +1572,9 @@ type Config struct {
 	// Subnets to place the service into. Defaults to the subnets in the default VPC.
 	Subnets []string `hcl:"subnets,optional"`
 
+	// SecurityGroups to pass security groups to deployment.
```
Comment (evanphx) [inline builtin/aws/ecs/platform.go:1575]: This should specify that it's the security group ids. Also, you need to add security_groups to the Documentation function below here.

**Review by evanphx — CHANGES_REQUESTED** [review]

File: builtin/aws/ecs/platform.go [hunk 4]
```diff
@@ -1567,6 +1572,9 @@ type Config struct {
 	// Subnets to place the service into. Defaults to the subnets in the default VPC.
 	Subnets []string `hcl:"subnets,optional"`
 
+	// Security Groups ARN to define existing security groups for ecs.
```
Comment (krantzinator) [inline builtin/aws/ecs/platform.go:1575]: Does the ARN work here? When I tested this I had to use the Security Group ID.

Comment (psihachina) [reply builtin/aws/ecs/platform.go]: Sorry, I'm confused about the concepts. Fixed

**Comment by krantzinator** [comment]
Thanks for the quick response! We need to do the below items as well to get this fully ready. I can do it if you like, let me know.
* Update the `doc.SetField` to:
```
"Security Group IDs to define existing security groups for ecs",
		docs.Summary(
			"list of existing groups to be specified for the ecs",
```

* Rebase this branch onto `main`
* run `make gen/website-mdx`

That should do it!

**Comment by psihachina** [comment]
@krantzinator It will probably be better if you do this because I have a `make get/website-mdx` error. 

image

**Pull request closed without merging**